soroban-test-wasms/wasm-workspace/loadgen/src/lib.rs

```diff
@@ -4,32 +4,32 @@ use soroban_sdk::{contract, contractimpl, Bytes, Env, U256};
 #[contract]
 pub struct Contract;
 
-const EXTEND_LEDGER: u32 = 5_000_000;
-
 #[contractimpl]
 impl Contract {
-    // This will write num_entries of size size_kilo_bytes. The initial entry will have a key of
-    // starting_index, the next key starting_index + 1...
-    pub fn write(e: Env, starting_index: u32, num_entries: u32, size_kilo_bytes: u32) {
-        if num_entries == 0 || size_kilo_bytes == 0 {
-            panic!("num_entries and size_kilo_bytes must be greater than 0");
+    // This will write num_write_entries of size size_kilo_bytes. The initial entry will have a key of
+    // u32(0), the next key u32(1), etc.
+    pub fn do_work(
+        e: Env,
+        guest_cycles: u64,
+        host_cycles: u64,
+        num_write_entries: u32,
+        size_kilo_bytes: u32,
+    ) -> U256 {
+        if size_kilo_bytes == 0 && num_write_entries != 0 {
+            panic!("size_kilo_bytes must be greater than 0");
         }
 
-        let slice = [0_u8; 1024];
+        let mut slice = [0_u8; 1024];
+        e.prng().fill(&mut slice);
         let mut bytes = Bytes::new(&e);
         for _ in 0..size_kilo_bytes {
             bytes.extend_from_slice(&slice);
         }
 
-        for i in starting_index..starting_index + num_entries {
+        for i in 0..num_write_entries {
```

Comment:
I just wonder why did you remove the start index? I guess it doesn't really matter now, but would matter for parallelization eventually (as we'd want to use different ranges of keys)

Reply:
Initially, I had one instance and had TXs use different key ranges like you mentioned. However, now that we have one instance per account, each invocation can use the same keys with no collisions. I didn't think about parallelization when I took it out, but that's still pretty far down the road.

Reply:
Ah, I see. Well, I guess then for parallelization we will want a bit of write key collisions, but I agree, that definitely can wait.